homeassistant/helpers/entity_registry.py

```diff
@@ -300,11 +300,9 @@ def async_generate_entity_id(
             or not self.hass.states.async_available(test_string)
         ):
             tries += 1
-            test_string = f"{preferred_string}_{tries}"
-
-        if len(test_string) > MAX_LENGTH_STATE_ENTITY_ID:
-            raise MaxLengthExceeded(
```

Comment:
I agree with removing this. It makes no sense to raise based on a value that we just generated.

Reply:
Yes, also I couldn't trigger the exception after the fix so codecov was complaining.

There might be a theoretical exception if you have 10^254 entities with similar names.

Not enough energy in the universe to iterate that far though!